### PR TITLE
Fix arrow expression in record and object

### PR DESCRIPTION
**Pull request by Anoukh**
## Purpose
Fix assigning arrow expressions inside record and object 
```ballerina
type Foo record {
    function (int) returns int lambda = (i) => i * i;
};
```

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java [hunk 1]
```diff
@@ -562,9 +564,10 @@ public boolean isArrayOpenSealedType(BArrayType arrayType) {
      */
     private SymbolEnv findEnclosingInvokableEnv(SymbolEnv env, BLangInvokableNode encInvokable) {
         if (env.enclEnv.node != null && env.enclEnv.node.getKind() == NodeKind.ARROW_EXPR) {
+            // if enclosing env's node is arrow expression
             return env.enclEnv;
         }
-        if (env.enclInvokable == encInvokable) {
+        if (env.enclInvokable == encInvokable && env.enclInvokable != null) {
```
Comment (Kishanthan) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java]: Is this check correct? We are doing a null check after checking for an equivalency on the same?

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java [hunk 2]
```diff
@@ -513,7 +513,9 @@ public void visit(BLangSimpleVarRef varRefExpr) {
                         ((BLangFunction) env.enclInvokable).closureVarSymbols.add((BVarSymbol) closureVarSymbol);
                     }
                 }
-                if (env.node.getKind() == NodeKind.ARROW_EXPR) {
+                if (env.node.getKind() == NodeKind.ARROW_EXPR && !(symbol.owner instanceof BPackageSymbol)) {
+                    symbol.owner = Symbols.createInvokableSymbol(SymTag.FUNCTION, 0, null,
```
Comment (Kishanthan) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java]: Why does this new symbol creation needed? What is purpose for the new binary expr check? Shall we add comment on it.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java (hunk 1 above)
Comment (Anoukh) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java]: In arrow expression the `encInvokable` also becomes null. Makes sense to do the null check first though. Fixed in commit 321ba02

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java (hunk 2 above)
Comment (Anoukh) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java]: Added a comment explaining the change

**Pull request merged**